File: android/core/data/src/main/kotlin/com/devora/mencare/core/data/fake/InMemoryStore.kt
package com.devora.mencare.core.data.fake

import com.devora.mencare.core.model.AppNotification
import com.devora.mencare.core.model.Appointment
import com.devora.mencare.core.model.ClientNotificationPrefs
import com.devora.mencare.core.model.ClientRecord
import com.devora.mencare.core.model.Holiday
import com.devora.mencare.core.model.NotificationSettings
import com.devora.mencare.core.model.Operator
import com.devora.mencare.core.model.PushCampaign
import com.devora.mencare.core.model.Salon
import com.devora.mencare.core.model.Service
import com.devora.mencare.core.model.TimeBlock
import com.devora.mencare.core.model.User
import com.devora.mencare.core.model.WaitlistEntry
import kotlinx.coroutines.flow.MutableStateFlow
import java.time.LocalDateTime
import java.util.concurrent.atomic.AtomicLong
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Single source of truth for all fake repositories. Every mutation goes through
 * the [MutableStateFlow]s so any screen observing the data updates live.
 */
@Singleton
class InMemoryStore @Inject constructor() {

    val salon = MutableStateFlow(Salon("", "", ""))
    val services = MutableStateFlow<List<Service>>(emptyList())
    val operators = MutableStateFlow<List<Operator>>(emptyList())
    val holidays = MutableStateFlow<List<Holiday>>(emptyList())
    val users = MutableStateFlow<List<User>>(emptyList())
    val clients = MutableStateFlow<List<ClientRecord>>(emptyList())
    val appointments = MutableStateFlow<List<Appointment>>(emptyList())
    val waitlist = MutableStateFlow<List<WaitlistEntry>>(emptyList())
    val timeBlocks = MutableStateFlow<List<TimeBlock>>(emptyList())
    val campaigns = MutableStateFlow<List<PushCampaign>>(emptyList())
    val notificationSettings = MutableStateFlow(NotificationSettings(emptyList()))
    val notifications = MutableStateFlow<List<AppNotification>>(emptyList())

    val currentUser = MutableStateFlow<User?>(null)
    val clientPrefs = MutableStateFlow(ClientNotificationPrefs())

    private val counter = AtomicLong(1000)

    fun newId(prefix: String): String = "${prefix}_${counter.incrementAndGet()}"

    fun now(): LocalDateTime = LocalDateTime.now()

    init {
        DemoSeed.seed(this)
    }
}

File: ios/MenCare/Core/Data/Fake/InMemoryStore.swift
import Foundation
import Observation

/// Single source of truth for all fake repositories. Every property is
/// observation-tracked, so any screen reading the data updates live —
/// the SwiftUI counterpart of the Android `MutableStateFlow` store.
@Observable
final class InMemoryStore {

    var salon = Salon(name: "", address: "", city: "")
    var services: [Service] = []
    var operators: [Operator] = []
    var holidays: [Holiday] = []
    var users: [User] = []
    var clients: [ClientRecord] = []
    var appointments: [Appointment] = []
    var waitlist: [WaitlistEntry] = []
    var timeBlocks: [TimeBlock] = []
    var campaigns: [PushCampaign] = []
    var notificationSettings = NotificationSettings(reminders: [])
    var notifications: [AppNotification] = []

    var currentUser: User?
    var clientPrefs = ClientNotificationPrefs()

    @ObservationIgnored private var counter: Int64 = 1000

    func newId(_ prefix: String) -> String {
        counter += 1
        return "\(prefix)_\(counter)"
    }

    func now() -> LocalDateTime { .now() }

    init() {
        DemoSeed.seed(self)
    }
}
